src/query/interpreter.cpp

```diff
@@ -986,6 +1037,16 @@ PreparedQuery PrepareCypherQuery(ParsedQuery parsed_query, std::map<std::string,
     spdlog::info("Running query with memory limit of {}", utils::GetReadableSize(*memory_limit));
   }
 
+  if (const auto &clauses = cypher_query->single_query_->clauses_;
+      !clauses.empty() && clauses[0]->GetTypeInfo() == LoadCsv::kType) {
```

Comment:
why first clause? `LOAD CSV` can be anywhere in the query.

Reply:
Hmm I didn't know that, thanks for noticing